src/frontend/src/lib/components/views/RecoveryPhraseInput.svelte

```diff
@@ -1,59 +1,77 @@
 <script lang="ts">
   import Tooltip from "$lib/components/ui/Tooltip.svelte";
-  import { InfoIcon } from "@lucide/svelte";
+  import { EyeIcon, EyeOffIcon, InfoIcon, ListXIcon } from "@lucide/svelte";
   import { t } from "$lib/stores/locale.store";
   import { onMount } from "svelte";
+  import type { SvelteHTMLElements } from "svelte/elements";
+  import ProgressRing from "$lib/components/ui/ProgressRing.svelte";
 
-  interface Props {
-    value: string[];
-    showValues?: boolean;
-    disabled?: boolean;
-  }
+  const EMPTY_PHRASE = Array.from({ length: 24 }).map(() => "");
+
+  type Props = {
+    value?: string[];
+    onSubmit: (recoveryPhrase: string[]) => Promise<void>;
+  } & SvelteHTMLElements["div"];
 
   let {
-    value = $bindable(),
-    showValues = false,
-    disabled = false,
+    value: initialValue,
+    onSubmit,
+    class: className,
+    ...props
   }: Props = $props();
 
-  let words = $derived(Array.from(value));
-  let wrapperRef = $state<HTMLDivElement>();
+  let value = $state(initialValue ?? EMPTY_PHRASE);
+  let wrapperRef = $state<HTMLFieldSetElement>();
   let dictionary = $state<string[]>();
+  let isSubmitting = $state(false);
+  let showValues = $state(false);
 
-  const isCompleteAndCorrect = $derived(
-    words.every(
-      (word) =>
-        word.length > 0 &&
-        dictionary !== undefined &&
-        dictionary.includes(word),
+  const isEmpty = $derived(value.every((word) => word.length === 0));
+  const isValid = $derived(
+    value.every(
+      (word) => dictionary !== undefined && dictionary.includes(word),
     ),
   );
-  const hasChanges = $derived(words.join(" ") === value.join(" "));
   const inputPattern = $derived(dictionary?.join("|"));
-  const firstEmptyInputIndex = $derived(
-    words.findIndex((word) => word.length === 0),
-  );
 
-  /** Focus on input (if it can be found for given index) */
-  const focusInput = (index: number) =>
+  /**
+   * Focus on an input field, noop if not found
+   *
+   * @param index of the input field to focus
+   */
+  const focusInput = (index: number) => {
     wrapperRef?.querySelectorAll("input")[index]?.focus();
```

Comment:
could you clarify if it's always safe to call `.focus` on whatever is returned by `wrapperRef?.querySelectorAll("input")[index]?`?